dev/ProgressBar/ProgressBar.cpp

```diff
@@ -93,32 +94,47 @@ void ProgressBar::UpdateStates()
     }
     else if (!IsIndeterminate())
     {
-        SetProgressBarIndicatorWidth();
         winrt::VisualStateManager::GoToState(*this, s_DeterminateStateName, true);
     }
 }
 
 void ProgressBar::SetProgressBarIndicatorWidth()
 {
+    const auto templateSettings = winrt::get_self<::ProgressBarTemplateSettings>(TemplateSettings());
+
     if (auto&& progressBar = m_layoutRoot.get())
     {
         if (auto&& progressBarIndicator = m_progressBarIndicator.get())
         {
             const double progressBarWidth = progressBar.ActualWidth();
+            const double prevIndicatorWidth = progressBarIndicator.ActualWidth();
             const double maximum = Maximum();
             const double minimum = Minimum();
             const auto padding = Padding();
 
-            if (std::abs(maximum - minimum) > DBL_EPSILON)
+            m_isUpdating = true; // Adds "Updating" state in between to trigger RepositionThemeAnimation Visual Transition in ProgressBar.xaml when reverting back to previous state
+            UpdateStates();
```

Comment:
Alternatively you could just call `
        winrt::VisualStateManager::GoToState(*this, s_UpdatingStateName, true);`  This way you wouldn't need the m_isUpdating flag.

Reply:
Is it cleaner to just call `winrt::VisualStateManager::GoToState(*this, s_UpdatingStateName, true);` ?

Reply:
@kmahone your thoughts? I think so.

---
In reply to: [346482548](https://github.com/microsoft/microsoft-ui-xaml/pull/1564#discussion_r346482548) [](ancestors = 346482548)

Reply:
I think it could go either way. But directly calling GoToState is probably neater, since it avoids the requirement of adding a bool member variable.